mailpoet/lib/Automation/Integrations/WooCommerce/Fields/CustomerOrderFieldsFactory.php

```diff
@@ -69,7 +69,8 @@ function (CustomerPayload $payload) {
         function (CustomerPayload $payload) {
           $customer = $payload->getCustomer();
           if (!$customer) {
-            return null;
+            $order = $payload->getOrder();
+            return $order && $order->is_paid() ? $order->get_date_created() : null;
```

Comment:
Is there a reason you prefer to keep this logic in the factory instead of the CustomerPayload? There's a tiny amount of duplication below, and also this code also depends on the idea that a guest customer payload will only have a single order, which I want to double check is a valid assumption.

To be clear I think it's totally fine for this code to live here, more just curious about how you decided where it should live.

Reply:
I can't move this to the payload unless I move there all of the `$this->getPaidOrderDate($customer, true);` method implementation, and I'm unsure whether it should go there or not. It would be rather inconsistent with other factories. Also, in the factory I have access to DI, in the payload not.

> the idea that a guest customer payload will only have a single order, which I want to double check is a valid assumption

That will be the case as there is no entity for a guest customer in Woo, so the fact that it's a guest is only derived from the order (the order has no customer attached).